src/cpp/rtps/builtin/discovery/endpoint/EDP.cpp

```diff
@@ -473,6 +474,12 @@ bool EDP::unpairWriterProxy(
                 info.status = REMOVED_MATCHING;
                 info.remoteEndpointGuid = writer_guid;
                 (*rit)->getListener()->onReaderMatched((*rit),info);
+
+                eprosima::fastdds::dds::SubscriptionMatchedStatus sub_info;
+                sub_info.current_count--;
+                sub_info.current_count_change--;
+                sub_info.last_publication_handle = writer_guid;
+                (*rit)->getListener()->onReaderMatched((*rit), sub_info);
```

Comment:
Creating a local `SubscriptionMatchedStatus` causes that `sub_info.total_count` always is ser to 0.
We propose to keep a map of `SubscriptionMatchedStatus` with the reader/writer `guid` as the key and to keep track of it.